lib/msal-browser/src/naa/mapping/NestedAppAuthAdapter.ts

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { TokenRequest } from "../TokenRequest";
+import { AccountInfo as NaaAccountInfo } from "../AccountInfo";
+import { RedirectRequest } from "../../request/RedirectRequest";
+import { PopupRequest } from "../../request/PopupRequest";
+import { TokenResponse } from "../TokenResponse";
+import {
+    AccountInfo as MsalAccountInfo,
+    AuthError,
+    AuthenticationResult,
+    ClientAuthError,
+    ClientConfigurationError,
+    InteractionRequiredAuthError,
+    ServerError,
+    TimeUtils,
+    AuthToken,
+    ICrypto,
+    Logger,
+} from "@azure/msal-common";
+import { BridgeError, isBridgeError } from "../BridgeError";
+import { BridgeStatusCode } from "../BridgeStatusCode";
+import { SilentRequest } from "../../request/SilentRequest";
+
+export class NestedAppAuthAdapter {
+    protected crypto: ICrypto;
+    protected logger: Logger;
+    protected clientId: string;
+
+    constructor(clientId: string, crypto: ICrypto, logger: Logger) {
+        this.clientId = clientId;
+        this.crypto = crypto;
+        this.logger = logger;
+    }
+
+    public toNaaSilentTokenRequest(request: SilentRequest): TokenRequest {
+        let extraParams: Map<string, string>;
+        if (request.extraQueryParameters === undefined) {
+            extraParams = new Map<string, string>();
+        } else {
+            extraParams = new Map<string, string>(
+                Object.entries(request.extraQueryParameters)
+            );
+        }
+        /**
+         * Need to get information about the client to populate request correctly
+         * For example: client id, client capabilities
+         */
+        const tokenRequest: TokenRequest = {
+            userObjectId: request.account?.homeAccountId,
+            clientId: this.clientId,
+            authority: request.authority,
+            scope: request.scopes.join(" "),
+            correlationId:
+                request.correlationId !== undefined
+                    ? request.correlationId
+                    : "",
+            prompt: request.prompt !== undefined ? request.prompt : "",
+            claims: request.claims !== undefined ? request.claims : "",
+            authenticationScheme:
+                request.authenticationScheme !== undefined
+                    ? request.authenticationScheme
+                    : "",
+            clientCapabilities: undefined,
```

Comment:
For this, I think we pass the value provided by the caller [of MSAL.js], right? Doesn't the embedded app declare what its capabilities are?

Reply:
I think that's what we will have to do... since we don't have the declarative model yet.

Reply:
Done.